rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,139 @@
+
+  - The separate RIF files associated with future claim data for the given release will be put uploaded in a specially designated folder in S3 for staging until loading in the future. 
+
+  - Additional testing - As of 04/21/2022, there is a plan in place to remve tens of thousands of old synthetic beneficiary data prior to Synthea data being stored in the database, which is largely responsible for the database collisions that have taken place in the past, when loading new Synthea data. Once this data is permenantly removed, it will likely take many years for there to be enough data to cause overlap. So after making sure the data generated does not collide with the previous batch of Synthea data, it should be safe to load.  
+
+Large Synthetic Data Generation:
+  - Each PI (Program Increment), approximately every 8 weeks, 60 million Synthea beneficiaries will be generated and available for on-demand performance and load testing on TEST and PROD SBX databases. The number of beneficiaries in the dataset size is meant to reflect the size and shape of production. The process will be similar to that of automated recurring generation, however the memory and computational power of the AWS instance will be larger for time and cost effectiveness. 
+  
+  - Using a m5a.24xlarge EC2 instance, it took 6 hours to generate 1 million synthetic beneficiaries. If one extrapolates from this benchmark, it would take 60 hours or a little under 3 days to generate 10 million, and 360 hours, or 15 days to generate 60 million beneficiaries.
+
+Required BFD Application Changes:
+When CCW data is ingested by the BFD pipeline, the application has filter logic for beneficiary UPDATEs with reference years prior to the current year by default. However, Synthea data must be able to UPDATE prior years. For both Synthea and CCW data to be ingested by the pipeline application, For both CCW and Synthea data to be ingested, the back-dated beneficiary filtering will have to be turned off for Synthea data. 
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Automated Generation & Load Plan:
+* On a weekly basis a cron job within Jenkins will be set up to start up and SSH into a provisioned AWS EC2 instance that will be turned off when not used, to run an Ansible script that takes a batch size and randomly generated year between 1 and 5 and will execute other scripts that:
+  - Run currently manual steps in the `Synthea Test Plan` such as accessing and comparing the latest Synthea end-properties file, running queries in PROD SBX to determine database ranges for beneficiary and claim properities, and ensuring the next batch of data will not overlap with existing data ranges.
```

Comment:
Where will the end-properties file be saved/read from?

Reply:
AWS S3. This is stated later in the detailed design. I tried to be minimal with discussing architecture in this section.